packages/grafana-ui/src/components/Box/Box.tsx

```diff
@@ -0,0 +1,113 @@
+import { css } from '@emotion/css';
+import React, { ElementType, useCallback } from 'react';
+
+import { GrafanaTheme2, ThemeSpacingTokens } from '@grafana/data';
+
+import { useStyles2 } from '../../themes';
+interface BoxProps {
+  margin?: ThemeSpacingTokens;
+  marginX?: ThemeSpacingTokens;
+  marginY?: ThemeSpacingTokens;
+  marginTop?: ThemeSpacingTokens;
+  marginBottom?: ThemeSpacingTokens;
+  marginLeft?: ThemeSpacingTokens;
+  marginRight?: ThemeSpacingTokens;
+
+  padding?: ThemeSpacingTokens;
+  paddingX?: ThemeSpacingTokens;
+  paddingY?: ThemeSpacingTokens;
+  paddingTop?: ThemeSpacingTokens;
+  paddingBottom?: ThemeSpacingTokens;
+  paddingLeft?: ThemeSpacingTokens;
+  paddingRight?: ThemeSpacingTokens;
+
+  backgroundColor?: keyof GrafanaTheme2['colors']['background'];
+  display?: 'flex' | 'block' | 'inline' | 'none';
+  element?: ElementType;
+}
+
+export const Box = ({ children, ...props }: React.PropsWithChildren<BoxProps>) => {
+  const styles = useStyles2(useCallback((theme) => getStyles(theme, props), [props]));
```

Comment:
Is this memoization not pointless?  Props will always be a new object instance on every render (when triggered via parent render)

I think the individual prop values needs to be in the dependency array for it to work, but could be wrong